test/extended/node_tuning/node_tuning_utils.go

```diff
@@ -0,0 +1,238 @@
+package node_tuning
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const masterNodeRoleLabel = "node-role.kubernetes.io/master"
+
+// isSNOCluster will check if OCP is a single node cluster
+func isSNOCluster(oc *exutil.CLI) bool {
+	infrastructureType, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	e2e.Logf("the cluster type is %v", infrastructureType.Status.ControlPlaneTopology)
+	return infrastructureType.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode
+}
+
+func getFirstMasterNode(ctx context.Context, nodeClient v1.NodeInterface) (*corev1.Node, error) {
+	masterNodes, err := nodeClient.List(ctx, metav1.ListOptions{LabelSelector: masterNodeRoleLabel})
+	if err != nil {
+		fmt.Printf("failed to list master nodes:'%v'", err)
+	}
+	var firstMasterName *corev1.Node
+	for i, masterNode := range masterNodes.Items {
+		if i == 0 {
+			firstMasterName = &masterNode
+			e2e.Logf("the first masterNode is %v", masterNode.Name)
+			break
+		}
+	}
+	return firstMasterName, err
+}
+
+func assertPodLogsLastLines(podName string, podLogs string, filter string) bool {
+	regNTOPodLogs, err := regexp.Compile(".*" + filter + ".*")
```

Comment:
Why do we need '.*' here?  Shouldn't:
'filter' and '".*" + filter + ".*"' match exactly the same strings?  If it does, code will be more robust with out that, because your "filter" then can be more flexible.

Reply:
Fixed in new code, thanks!